src/app/api/file/route.ts

```diff
@@ -0,0 +1,362 @@
+          filetype: fileData.filetype,
+          url: `https://docs.google.com/document/d/${googleFileId}`,
+          seniorId: fileData.seniorId,
+          Tags: fileData.Tags,
+        },
+      });
+
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "SUCCESS",
+          message: "File successfully added",
+        }),
+        { status: 200 }
+      );
+    }
+  } catch (e) {
+    console.log("Error:", e);
+    return NextResponse.json(
+      FileResponse.parse({
+        code: "UNKNOWN",
+        message: "Unknown error received",
+      }),
+      { status: 500 }
+    );
+  }
+});
+
+export const PATCH = withSession(async (request) => {
+  try {
+    const { service, fileRequest } = await initializeDriveAuth(request);
+
+    if (!fileRequest.success) {
+      console.log(fileRequest.error);
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "NOT_FOUND",
+          message: "Unsuccessful request creation",
+        }),
+        { status: 400 }
+      );
+    } else {
+      const fileData = fileRequest.data;
+
+      // Check that user has this senior assigned to them
+      const { SeniorIDs } = await prisma.user.findFirst({
+        where: {
+          id: request.session.user.id,
+        },
+      });
+
+      if (
+        !SeniorIDs.some((seniorId: string) => seniorId === fileData.seniorId)
+      ) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_AUTHORIZED",
+            message: "Senior not assigned to user",
+          }),
+          { status: 404 }
+        );
+      }
+
+      // Get senior from database
+      const foundSenior = await prisma.senior.findUnique({
+        where: { id: fileData.seniorId },
+      });
+      if (foundSenior == null) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_FOUND",
+            message: "Senior not found",
+          }),
+          { status: 404 }
+        );
+      }
+
+      const formatted_date =
+        (fileData.date.getMonth() > 8
+          ? fileData.date.getMonth() + 1
+          : "0" + (fileData.date.getMonth() + 1)) +
+        "/" +
+        (fileData.date.getDate() > 9
+          ? fileData.date.getDate()
+          : "0" + fileData.date.getDate()) +
+        "/" +
+        fileData.date.getFullYear();
+
+      // Used for extracting out the fileId from the url
+      const pattern =
+        /https:\/\/docs\.google\.com\/document\/d\/([a-zA-Z0-9_-]+)/;
+      const matches = pattern.exec(fileData.url);
+
+      if (matches && matches[1]) {
+        const googleFileId = matches[1];
+
+        const body = { name: formatted_date };
+
+        const fileUpdateData = {
+          fileId: googleFileId,
+          resource: body,
+        };
+
+        await (service as NonNullable<typeof service>).files.update(
+          fileUpdateData
+        );
+
+        const { id } = await prisma.file.findFirst({
```

Comment:
You might potentially run into the same problem here!

Reply:
Updated!!